### PR TITLE
Raise default timeout from 2s to 5s

### PR DESCRIPTION
**Pull request by IagoAbal**
Ran the full compare-semgrep benchmark and it did not show meaningful differences. In a sense, that means 5s does not seem to hurt perf and it also gives Semgrep extra flexibility when it is needed. Based on user/customer observations, we also believe it should be a better timeout. Let's try.

Helps FIND-1270

### CODE REVIEW

File: src/osemgrep/cli_scan/Scan_CLI.ml [hunk 1]
```diff
@@ -109,7 +109,7 @@ let default : conf =
          * not overload on large machines
          *)
         Core_runner.num_jobs = min 16 (Parmap_helpers.get_cpu_count ());
-        timeout = 2.0 (* seconds *);
+        timeout = 3.0 (* seconds *);
```
Comment (aryx) [inline src/osemgrep/cli_scan/Scan_CLI.ml]: please add a coupling: comment saying if you modify here, you need to modify constants.py and User_settings.ml

**Review by aryx — APPROVED** [review]

File: cli/src/semgrep/constants.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 DEFAULT_SEMGREP_CONFIG_NAME = "semgrep"
 DEFAULT_SEMGREP_APP_CONFIG_URL = "api/agent/deployments/scans/config"
 
-DEFAULT_TIMEOUT = 2  # seconds
+DEFAULT_TIMEOUT = 5  # seconds
```
Comment (chmccreery) [inline cli/src/semgrep/constants.py]:
```suggestion
DEFAULT_TIMEOUT = 5  # seconds, keep up-to-date with Scan_CLI.ml and User_settings.ml
```

File: src/osemgrep/cli_scan/Scan_CLI.ml [hunk 3]
```diff
@@ -109,7 +109,7 @@ let default : conf =
          * not overload on large machines
          *)
         Core_runner.num_jobs = min 16 (Parmap_helpers.get_cpu_count ());
-        timeout = 2.0 (* seconds *);
+        timeout = 5.0 (* seconds *);
```
Comment (chmccreery) [inline src/osemgrep/cli_scan/Scan_CLI.ml]:
```suggestion
        timeout = 5.0 (* seconds, keep up-to-date with User_settings.ml and constants.py *);
```

File: src/osemgrep/language_server/server/User_settings.ml [hunk 4]
```diff
@@ -47,7 +47,7 @@ let default =
     jobs = 1;
     max_memory = 0;
     max_target_bytes = 1000000;
-    timeout = 2;
+    timeout = 5;
```
Comment (chmccreery) [inline src/osemgrep/language_server/server/User_settings.ml:50]:
```suggestion
    timeout = 5; (* seconds, keep up-to-date with Scan_CLI.ml and constants.py *)
```

File: src/osemgrep/language_server/server/User_settings.ml [hunk 5]
```diff
@@ -47,7 +47,7 @@ let default =
     jobs = 1;
     max_memory = 0;
     max_target_bytes = 1000000;
-    timeout = 2;
+    timeout = 5; (* seconds, keep up-to-date with Scan_CLI.ml and constants.py *)
```
Comment (chmccreery) [inline src/osemgrep/language_server/server/User_settings.ml]:
```suggestion
    timeout = 5 (* seconds, keep up-to-date with Scan_CLI.ml and constants.py *);
```

**Pull request merged**